pkg/polaris/aws/aws.go

```diff
@@ -929,3 +929,148 @@ func (a API) TrustPolicies(ctx context.Context, id IdentityFunc, features []core
 
 	return trustPolicies, nil
 }
+
+func (a API) CreateCloudCluster(ctx context.Context, id uuid.UUID, input aws.CreateAwsClusterInput, useLatestCdmVersion bool) error {
+	a.log.Print(log.Trace)
+
+	// Ensure account exists and has Server and Apps feature
+	account, err := a.AccountByID(ctx, core.FeatureServerAndApps, id)
+	if err != nil {
+		return err
+	}
+
+	// If CloudAccountID is not specified, use the account ID from the account lookup
+	if input.CloudAccountID == uuid.Nil {
+		input.CloudAccountID = account.ID
+	}
+
+	// validate region in input
+	inputRegion := aws.RegionFromName(input.Region)
+
+	// Get Available CDM versions
+	cdmVersions, err := aws.Wrap(a.client).AllAwsCdmVersions(ctx, input.CloudAccountID, inputRegion)
+	if err != nil {
+		return fmt.Errorf("failed to get cdm versions: %s", err)
+	}
+
+	// Validate CDM version is available
+	validCdmVersion := false
+	supportedInstanceTypes := []aws.AwsCCInstanceType{}
+	if useLatestCdmVersion {
+		for _, version := range cdmVersions {
+			if version.IsLatest {
+				input.VmConfig.CdmVersion = version.Version
+				input.VmConfig.CdmProduct = version.ProductCodes[0]
+				supportedInstanceTypes = version.SupportedInstanceTypes
+				validCdmVersion = true
+				break
+			}
+		}
+	} else {
+		for _, version := range cdmVersions {
+			if version.Version == input.VmConfig.CdmVersion {
+				supportedInstanceTypes = version.SupportedInstanceTypes
+				input.VmConfig.CdmProduct = version.ProductCodes[0]
+				validCdmVersion = true
+				break
+			}
+		}
+	}
```

Comment:
nit - ought to be same as the below? or keep it if you think it is more readable as is
```
for _, version := range cdmVersions {
	if (version.IsLatest && useLatestCdmVersion) || (version.Version == input.VmConfig.CdmVersion) {
		validCdmVersion = true
		input.VmConfig.CdmVersion = version.Version
		input.VmConfig.CdmProduct = version.ProductCodes[0]
		supportedInstanceTypes = version.SupportedInstanceTypes
		break
	}
}